### PR TITLE
chore: adds unit tests and format github action

### PR DESCRIPTION
**Pull request by dutterbutter**
**Changes introduced in this PR:**

- Adds unit test for MyPaymaster.sol and MyERC20.sol 
- Adds checks.yml to CI for formatting and link checker
- Updates README.md
- Formatted 

To run tests follow the steps to start local test environment [here](https://era.zksync.io/docs/tools/testing/dockerized-testing.html) and then run `yarn test`. 

Closes DEVRL-29

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    strategy:
+      matrix:
+        node-version: [18.x]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: prettier
```
Comment (MexicanAce) [inline .github/workflows/checks.yml:31]: New Goal: Have tutorials run against In-Memory Node

File: test/MyPaymaster.test.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    );
+
+    await erc20Contract.estimateGas.mint(user.address, 5, {
+      customData: {
+        gasPerPubdata: utils.DEFAULT_GAS_PER_PUBDATA_LIMIT,
+        paymasterParams: paymasterParams,
+      },
+    });
+
+    await (
```
Comment (MexicanAce) [inline test/MyPaymaster.test.ts:73]: Do we need a double `await` here?

File: test/MyPaymaster.test.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    await expect(
+      executeTransaction(userWallet, "ApprovalBased", invalidTokenAddress),
+    ).to.be.rejectedWith("failed pre-paymaster preparation");
+  });
+
+  it("should revert if allowance is too low", async function () {
+    const erc20Contract = await getToken(userWallet);
+    await fundAccount(wallet, userWallet.address, "13");
+    await erc20Contract.approve(paymaster.address, ethers.BigNumber.from(0));
+    try {
```
Comment (MexicanAce) [inline test/MyPaymaster.test.ts:113]: I think with `chai-as-promised` you can simplify this `try/catch` to the following (if you want):
```javascript
await expect(executeTransaction(userWallet, "ApprovalBased", erc20.address.toString()))
    .to.be.rejectedWith("Min allowance too low");
```

File: test/MyERC20.test.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { expect } from "chai";
```
[the remark this answers is not in the document]
Comment (dutterbutter) [reply test/MyERC20.test.ts:1]: I don't think so, if we have another repo that includes a frontend we will separate the projects accordingly so `tests` will live in their respected project directory. If we move the `/tests` dir into `contracts` then the deploy script and hardhat config should follow and then you have this weird project structure imo.

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**